extension-base/src/main/java/com/azure/autorest/extension/base/plugin/JavaSettings.java

```diff
@@ -639,6 +639,27 @@ public boolean shouldUseIterable() {
         return serviceVersions;
     }
 
+    private Map<String, String> llcProperties;
+
+    public Map<String, String> getLlcProperties() {
+        if (llcProperties == null) {
+            llcProperties = new HashMap<>();
+        }
+        llcProperties.put("artifact-id", getArtifactId());
+        llcProperties.put("service-name", getServiceName());
+        llcProperties.put("namespace-path", getPackage().replaceAll("\\.", "/"));
```

Comment:
Is there a reason to use a `Map` instead of defining a Java class for all the supported properties?

Reply:
Because there are large number of parameters and they are only used to render resource file template.